### PR TITLE
Indicate when a layer has two or more styles

### LINKED ISSUE #43087: Indicate when a layer has two or more styles
**Feature description.**
In the Layers panel would it be possible to decorate a layer item to show when the layer has more than one style defined.  Like the way the context menu has an option to `"Show Feature Count"`.  A little icon or something to show there are multiple styles available (and to switch between them), see the attached image, optionA.png.

optionA

Another option that wouldn't involve cluttering the Layers Panel:  Add the style count to the `Style `item in the conext menu, see the attached image, optionB.png.
optionB

**Additional context**
I often find myself making new styles for a layer to use in a Layout.  Then I kind of forget I have two styles and later I end up modifying the style and messing with the Layout.   If I had a visual indicator there were multiple styles I would be less likely to edit the wrong one.  The second option would also work for me because the Styles item is directly above the Properties item in the context menu so I would notice it when opening the Properties window.

Thanks

### PR DESCRIPTION
**Pull request by ismailsunni**
## Description

Indicate when a layer has two or more styles on the layer context menu (fix #43087)

Peek 2022-02-21 09-56 - show number of styles

Funded by Camptocamp https://www.camptocamp.com

### CODE REVIEW

File: src/app/qgsapplayertreeviewmenuprovider.cpp [hunk 1]
```diff
@@ -591,6 +592,11 @@ QMenu *QgsAppLayerTreeViewMenuProvider::createContextMenu()
       {
         menu->addSeparator();
         QMenu *menuStyleManager = new QMenu( tr( "Styles" ), menu );
+        QgsMapLayerStyleManager *mgr = layer->styleManager();
+        if ( mgr->styles().count() > 1 )
+        {
+          menuStyleManager->setTitle( tr( "Styles [%1 styles defined]" ).arg( mgr->styles().count() ) );
```
Comment (nyalldawson) [inline src/app/qgsapplayertreeviewmenuprovider.cpp:598]: What do you think about just "Styles [2]"? The extra "styles defined" text seems extraneous to me.

Comment (nyalldawson) [reply src/app/qgsapplayertreeviewmenuprovider.cpp:598]: Actually that should probably be "Styles (2)" to match other parts of QGIS

Comment (ismailsunni) [reply src/app/qgsapplayertreeviewmenuprovider.cpp]: Yes, it seems Styles (2) is much better. Updated now. Thanks, Nyall.

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Unrelated test failure

**Pull request merged**